### ISSUE TITLE
Fast access to channel data

### ISSUE THREAD

**Issue by dy** [audio-lab/buffer#13]
Working with [audio](https://github.com/audiojs/audio/tree/class) oftentimes there is `audio.read(time).getChannelData(channel)` to get raw array - pretty long. Is there a sense to provide id-access to channel to make `audio.read(time)[channel]`? @jamen

We could do that by initializing object properties on AudioBuffer instance.
```js
Object.defineProperties(audioBuffer, {
0: {
get: function () {return this.getChannelData(0)}
},
...
})
```

Or is it better to patch audioBuffer by audio itself? It is going to do that every `read` call, pretty slow.

**Comment by jamen** [comment]
I like the idea of a shorter way to access that data

But two questions I think of:

 - Would we keep the two ways to do the same thing? (Does that show some redundancy?)
 - Is using `audioBuf[int]` clear if the reference doesn't have `channel` in it?  I feel like it could be mistook for pulse, but maybe not.

Agree with the doing it every `read` call would be slow.  Allowing syntax from audio and keeping audio-buffer with less forgiving API to the data structure makes sense to me.

**Comment by dy** [comment]
Ok, audio-buffer is low-level enough structure to keep it sugar-free.

**Comment by dy** [comment]
Although what if we rename `audioBuffer.data` to `audioBuffer.channels`?
* :white_check_mark: we would fix that agreement in API, exposing channel data and number of channels naturally, anyways we use array with channels data and unlikely will get back to single-row array.
* :white_check_mark: that is natural pattern from [audio](https://github.com/audiojs/audio) - to pass channels array
* :white_check_mark:  that would start making sense for channels argument as `[lData, rData]`
* :negative_squared_cross_mark: that is not compliant with WAA AudioBuffers, we'd wave to wrap them, although we already non-compatible due to constructor.
* :negative_squared_cross_mark: performance-wise it potentially may be slower than pure audio-buffer, although in long-term that can be winning due to exposed `getChannelData(N)`
* :negative_squared_cross_mark: in case if we return native AudioBuffer, it will not have that property, although we can define it on the created native buffer − how bad is that? For @audiojs infrastructure that is win, for compatibility that is loss

**Comment by dy** [comment]
Fixed in audio-buffer@4.0.0
